crates/driver/example.toml

```diff
@@ -4,6 +4,7 @@ endpoint = "http://0.0.0.0:7872"
 absolute-slippage = "40000000000000000" # Denominated in wei, optional
 relative-slippage = "0.1" # Percentage in the [0, 1] range
 account = "0x0000000000000000000000000000000000000000000000000000000000000001" # The private key of the solver
+authorization = "FAKE_AUTH_TOKEN"
```

Comment:
I think something more generic might be nice. You could have a map instead that maps header to value in case different solvers expect the auth token to be passed using different headers.

Reply:
you mean like a mapping that says if the toml key is "authorization" or "auth_token" or any other then it should be interpreted as the authorization key for the config? What would the benefit of that be and would it not add more unnecessary complexity?

Reply:
I mean a simple mapping `HEADER -> VALUE` and for every request we send we set the header `HEADER` to `VALUE`.
That way we could support any different header solvers might expect. `Autothorization`, `X-API-KEY` or whatever else they decided to listen on.
This might also come in handy when solvers only want to expose a single `/solve` endpoint for multiple networks. Then we could for example set `NETWORK` to `mainnet` with the same solution.

Reply:
Actually one of the solver team even uses URL query parameters in a similar fashion so you could also add that while you are at it. Should mostly be copy-and-paste.

Reply:
Do you want to be able to support any arbitrary header or just arbitrary names for the authorization header. The former of those 2 is looking like it might be rather complex.

Reply:
I want to support that we can configure a mapping like:
```
header1 => value1,
header2 => value2,
//...
```
That way we can configure any static values and authorization wouldn't even be a special case. Since headers can be any string I think this should actually be less amount of work since you can just apply the headers on each request without having to map it to an authorization header specifically.
So in the case of the authorization token the mapping would look like:
```
"Authorization" => "$API_KEY"
```

Reply:
Y'all are obviously more familiar with your code than I am but wouldn't you need to map a key to a specific header? The header insertion requires an enum and a value so you would need to map to both the header name and the header value. 
`        headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());`

Reply:
Ahh, now I see where the confusion comes from.
`.insert()` does not actually only take an enum. Instead you can pass anything that implements `IntoHeaderName`. This is implemented for the header enum but also for static strings.
However, static strings will also not do us any good so we can not use `reqwest::header::HeaderMap` like we do at the moment but instead have to add each header individually with `.header()` on the request itself (e.g. like [here](https://github.com/cowprotocol/services/blob/main/crates/shared/src/paraswap_api.rs#L57-L60)).
That way we can populate header names with owned `String`s which makes parsing them from a file easy.

Reply:
Sure so that makes sense on the header name part, but the parsing of the toml file still presents some challenges because you're parsing contents into Structs `SolverConfig` [here](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/config/file/load.rs#L51), which has rigidly defined fields already. I'm not sure how you would go about modifying the SolverConfig struct to represent all other fields. I'm sure there's a way but Idk what that is.

Reply:
You only have to add sth like:
```rust
    #[serde(default)]
    request_headers: HashMap<String, String>,

    #[serde(default)]
    request_query_parameters: HashMap<String, String>,
```
where the config file gets parsed [[0](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/config/file/mod.rs#L52)] and go from there.

And then you can set values in the config like:
```
[solver.request-headers]
greeting = "moo"
Authorization = "TOP_SECRET"
```

Reply:
Ah ok sure yea just make different toml sections that's more reasonable. I was interpreting it as you wanted it all to be under the general solver header so it had to be mixed with the other mandatory ones. Yea that seems not too bad. I'll take a crack at it see what I can put together. Thanks for the clarification

Reply:
So if i'm understanding this code correctly, at [this](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/config/file/load.rs#L51) line, the join_all starts iterating over the vec of solverConfigs, but inside the closure there's a move using the same name config, but a different type (SolverConfig vs file::Config). If the additional headers are stored in the file::config but `config` points to a type `solverConfig`, how should I go about retrieving the values in the hashmap that were stored in the file::Config object [here](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/config/file/load.rs#L30)? I guess this may be more of a rust question than a code specific question

Reply:
I've pushed up a change that should work. I ran the `example_config` test and it seemed to be working.